sklearn/utils/fixes.py

```diff
@@ -402,3 +407,21 @@ def _in_unstable_openblas_configuration():
             # See discussions in https://github.com/numpy/numpy/issues/19411
             return True  # pragma: no cover
     return False
+
+
+# TODO: remove when pandas >= 1.4 is the minimum supported version
+if pd is not None and parse_version(pd.__version__) < parse_version("1.4"):
+
+    def _create_pandas_dataframe_from_non_pandas_container(X, *, index, copy):
+        X_output = pd.DataFrame(X, index=index, copy=copy)
+        if "polars" not in str(X.__class__):
```

Comment:
Is it good enough to test `"polars" in class` or should we check the more specific
```suggestion
        if str(X.__class__).startswith("polars."):
```

Reply:
Fair question: originally I wanted to avoid importing `polars` to check `isinstance(X, pl.DataFrame)` I don't really remember why if to be fully honest :thinking:. There is also `sklearn.util.validation._is_polar_df` although this will likely cause a circular import, `sklearn.utils.validation` imports stuff from `sklearn.utils.fixes` so it's not OK to use import `sklearn.utils.validation` in `sklearn.utils.fixes` reverse.

Let me try to think a bit more about this.

Reply:
I pushed some better code in https://github.com/scikit-learn/scikit-learn/pull/29502/commits/fdf23edd9f80098bf582c7e1fb1594e680fecabd